FHEM/10_FS10.pm

```diff
@@ -1,502 +1,556 @@
+
+	my $setstate = $a[0];
+	my $ebeneh = substr $hash->{BTN}, 0, 1;
+	my $ebenel = substr $hash->{BTN}, 1, 1;
+	my $housecode = $hash->{HC} - 1;
+	my $kc;
+	my $SignalRepeats = AttrVal($name,'repetition', '0') + 1;
+	my $dimm = 0;
+	my $newmsg = 'P61#';
+
+	if ($model eq 'FS10_MS') {
+		$SignalRepeats = 1;
+	}
+
+	if ($SignalRepeats > 10) {
+		$SignalRepeats = 10;
+	}
+
+	if ($na > 1 && $setstate =~ m/dim/xms) { # Anzahl dimup / dimdown
+		$dimm += $a[1];
+		if ($dimm < 1 || $dimm > 10) {
+			Log3 $name, 1, "$ioname: FS10 set $name $setstate $dimm - ERROR dimm value too low or high (1-10)";
+			return "FS10 set $name $setstate $dimm - ERROR: dimm value too low or high (1-10)";
+		} else {
+			Log3 $name, 3, "$ioname: FS10 set $name $setstate $dimm $alias";
+		}
+	} else {
+		Log3 $name, 3, "$ioname: FS10 set $name $setstate $alias";
+	}
+	Log3 $name, 5, "$ioname: FS10 set $name hc=$housecode ebeneHL=$ebeneh$ebenel setstate=$setstate";
+
+	for my $i (1..2) {
+		my $sum = 0;
+		$kc = $fs10_c2b{$setstate . '_' . $i};
+		$kc = $kc & 7;
+		if (defined $kc) {
+			Log3 $name, 5, "$ioname: FS10 set $name setstate$i=$setstate command=$kc";
+			$newmsg .= '0000000000001'; # 12 Bit Praeambel, 1 Pruefbit
+			$newmsg .= dec2nibble($kc); # 1. setstate
+			$sum += $kc;
+			$newmsg .= dec2nibble($ebenel); # 2. Ebene low
+			$sum += $ebenel;
+			$newmsg .= dec2nibble($ebeneh); # 3. Ebene high
+			$sum += $ebeneh;
+			$newmsg .= '10001'; # 4. unused
+			$newmsg .= dec2nibble($housecode); # 5. housecode
+			$sum += $housecode;
+			$sum = (10 - $sum) & 7;
+			$newmsg .= dec2nibble($sum); # 6. Summe
+			if ($dimm == 0) { # ein / aus
+				if ($i == 1) { # 1. Teil Nachricht
+					$newmsg .= 'PPP'; # 3*32400=97200 Pause
+				} else { # 2. Teil Nachricht
+					if ($SignalRepeats == 1) {
+						$newmsg .= '#R1'; # 1 Repeat
+					} else {
+						$newmsg .= 'PPPPPP#R' . $SignalRepeats; # 6*32400=194400 Pause . Repeats
+					}
+				}
+			} else { # dimmen
+				if ($i == 1) { # 1. Nachricht
+					$newmsg .= 'PPPPPPPPPPPPPPPP'; # 16*32400=518400 Pause . 1 Repeat (original remote control)
+					if ($dimm >= 2) {
+						$newmsg .= '#R1';
+						IOWrite($hash, 'sendMsg', $newmsg);
+						Log3 $name, 5, "$ioname: FS10 set dimm $dimm, 1. sendMsg=$newmsg";
+						$newmsg = 'P61#'; # Reset newmsg fuer 2. Nachricht
+					}
+				} else { # 2. Nachricht
+					if ($dimm == 1) {
+						$newmsg .= '#R1'; # 1 Repeat
+					} else {
+						$newmsg .= 'PPPPPP#R' . $dimm; # 6*32400=194400 Pause . Repeats
+						Log3 $name, 5, "$ioname: FS10 set dimm $dimm, 2. sendMsg=$newmsg";
+					}
+				}
+			}
+			if ($i == 2) { # 2. Nachricht
+				Log3 $name, 5, "$ioname: FS10 set sendMsg=$newmsg";
+				IOWrite($hash, 'sendMsg', $newmsg);
+			}
+		}
+	}
+
+	# Set the state of a device to off if on-for-timer is called
+	if ($modules{FS10}{ldata}{$name}) {
+		CommandDelete(undef, $name . '_timer');
+		delete $modules{FS10}{ldata}{$name};
+	}
+
+	# following timers
```

Comment:
Diese following Timers,

Was sollen die eigentlich machen?
War vorher schon unklar, aber ich habe nichts gefunden in der commandref

Reply:
Findest du unter "follow-on-timer" in der commandref und stammt ursprünglich aus FS20.
Man kann wohl bei einigen Funkschaltern einen Timer programmieren, so das das Gerät nach Ablauf der Zeit selbstständig ausschaltet. Weil FHEM davon nichts mitbekommt, kann man mit diesem Attribut die Zeit festlegen, nach der FHEM den state wechselt.

Reply:
Ist mir ehrlich gesagt noch nie unter gekommen.

Passt nicht in diesen pr, aber der Standard Befehlsumfang ist über setextensions einbindbar :)

Reply:
Das ist sicher auch ziemlich speziell. Die setextensions werden unterstützt.